### PR TITLE
license MIT -> apache 2.0

### PR DESCRIPTION
**Pull request by alk-lbinet**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -31,5 +31,5 @@
     zip_safe=False,
     install_requires=install_requires,
     setup_requires=["pytest-runner"],
-    license="MIT",
+    license="Apache Software License (http://www.apache.org/licenses/LICENSE-2.0)",
```
Comment (sethmlarson) [inline setup.py:34]: Typically this field would be the SPDX identifier for the license which is `"Apache-2.0"`

**Comment by sethmlarson** [comment]
Thanks much, @alk-lbinet!

Another thing that you can do to be more explicit about the origin of certain code is to put a reference to the original project within code files that contain code taken from other places. You might consider doing this in addition to switching the license.

**Comment by alk-lbinet** [comment]
@sethmlarson I tried to do so when I found it relevant, for instance:
https://github.com/alkemics/pandagg/blob/master/pandagg/connections.py#L1
https://github.com/alkemics/pandagg/blob/master/pandagg/search.py#L1
https://github.com/alkemics/pandagg/blob/master/pandagg/utils.py#L6

If I missed some spots it's unintentional, don't hesitate to ping me. 
Other inspirations came from the use of metaclass as a whole, or the cloning logic you might find in underlying lighttree library [here](https://github.com/leonardbinet/lighttree/blob/master/lighttree/tree.py#L116) , but it's less obvious to reference it

**Comment by sethmlarson** [comment]
Will do, thanks much!

**Pull request merged**